### PR TITLE
KEY-148: Field demos

### PR DESCRIPTION
**Pull request by jtbrolin**
- Adds interactive field demos to the Fields API pages (note: only the ones that were cheap to add at this stage)
- Adds info about `slugField` to `/docs/fields/slug`, `/docs/fields/array`, and `/docs/collections-and-singletons`

NOTE: 
- I also had to move some things around to make the Voussoir styles work correctly:
-- The font is now implemented as per https://nextjs.org/docs/app/building-your-application/optimizing/fonts#with-tailwind-css
-- The @tailwind/forms global styles were clashing with Voussoir, so now only importing the class names which needs to be used on all non-Voussoir inputs on docs site

### CODE REVIEW

File: docs/src/components/document-renderer.tsx [hunk 1]
```diff
@@ -120,9 +129,9 @@ const getRenderers = (
     layout: ({ children }) => {
       return (
         <div className="grid gap-6 my-2 grid-cols-1 sm:grid-cols-2">
-          {children.map((element, i) => (
+          {children.map((element, index) => (
             <div
-              key={i}
+              key={index}
```
Comment (jtbrolin) [inline docs/src/components/document-renderer.tsx:134]: Changed for consistency w rest of file

File: docs/src/components/document-renderer.tsx [hunk 2]
```diff
@@ -97,7 +106,7 @@ const getRenderers = (
         return (
           <ol className="text-keystatic-gray-dark list-decimal list-inside mt-2">
             {children.map((child, index) => (
-              <li key={index} className="mb-2 leading-tight">
+              <li key={index} className="mb-2">
```
Comment (jtbrolin) [inline docs/src/components/document-renderer.tsx:109]: Regretted adding this cause it made the lines too tight and hard to read

File: docs/src/components/document-renderer.tsx [hunk 3]
```diff
@@ -80,7 +89,7 @@ const getRenderers = (
 
       return (
         <div
-          className="[&>pre]:whitespace-break-spaces [&>pre]:break-all [&>pre]:p-4 [&>pre]:rounded-lg [&>pre]:border [&>pre]:border-keystatic-gray [&>pre]:bg-keystatic-gray-light text-sm my-2"
+          className="[&>pre]:whitespace-break-spaces [&>pre]:break-all [&>pre]:px-6 [&>pre]:py-4 [&>pre]:rounded-lg [&>pre]:border [&>pre]:border-keystatic-gray [&>pre]:bg-white text-sm my-2"
```
Comment (jtbrolin) [inline docs/src/components/document-renderer.tsx:92]: Change code examples to have white bg, cause I felt it looks cleaner/fresher especially now that the visual field demos are added and have a light grey bg as well

File: docs/src/components/fields/integer.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  return (
+    <FieldDemoFrame>
+      <IntegerFieldInput
+        label="Age"
+        onChange={setValue}
+        autoFocus={false}
+        forceValidation={false}
+        description="The person's age"
+        validation={{ min: 0, max: 120 }}
```
Comment (ChrisLaneAU) [inline docs/src/components/fields/integer.tsx:18]: We'll need to update this for [Maria](https://en.wikipedia.org/wiki/Maria_Branyas) in 3.5 years 😉

File: docs/src/components/fields/multiselect.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+import { FieldDemoFrame } from './frame';
+import { MultiselectFieldInput } from '../../../../packages/keystatic/src/form/fields/multiselect/ui';
+import { useState } from 'react';
+
+export const MultiselectFieldDemo = () => {
+  const [value, setValue] = useState<readonly string[]>([
+    'surfing',
+    'basketball',
+    'music',
```
Comment (ChrisLaneAU) [inline docs/src/components/fields/multiselect.tsx:11]: And `'chess'`?

Comment (ChrisLaneAU) [reply docs/src/components/fields/multiselect.tsx:11]: Or what about moving the `options` up into an `options` const and then using grabbing the type from that? Something like this:

```
const options = [
  { label: 'Surfing', value: 'surfing' },
  { label: 'Basketball', value: 'basketball' },
  { label: 'Music', value: 'music' },
  { label: 'Chess', value: 'chess' },
] as const;

type Options = (typeof options)[number]['value'];
```

File: docs/src/content/pages/fields/array.mdoc [hunk 6]
```diff
@@ -51,6 +51,34 @@ complexArray: fields.array(
 ),
 ```
 
+## Slug field
+
+The array field's `slugField` option is useful to replace indexes normally in file paths for images or documents e.g.
```
Comment (ChrisLaneAU) [inline docs/src/content/pages/fields/array.mdoc:56]: Is there an example missing at the end?

File: docs/src/content/pages/fields/file.mdoc [hunk 7]
```diff
@@ -0,0 +1,30 @@
+---
+title: File field
+---
+{% fieldDemo field="file" /%}
+
+The `file` field is used to store a file. In the Admin UI it renders an file picker component.
```
Comment (ChrisLaneAU) [inline docs/src/content/pages/fields/file.mdoc:6]: Small typo

```suggestion
The `file` field is used to store a file. In the Admin UI it renders a file picker component.
```

File: docs/src/content/pages/fields/file.mdoc [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+In [local storage mode](/docs/local-vs-git-hub-storage-strategy#local-strategy) it will save it as a file in your repository. It [GitHub storage mode](/docs/local-vs-git-hub-storage-strategy#git-hub-strategy) it will commit it into the repository.
```
Comment (ChrisLaneAU) [inline docs/src/content/pages/fields/file.mdoc:8]: What do you think about some slight copy changes for readability?

```suggestion
In [local storage mode](/docs/local-vs-git-hub-storage-strategy#local-strategy) the file will be saved in your repository. In [GitHub storage mode](/docs/local-vs-git-hub-storage-strategy#git-hub-strategy) the file will be committed into the repository.
```

File: docs/src/content/pages/fields/slug.mdoc [hunk 9]
```diff
@@ -23,6 +25,17 @@ title: fields.slug({
 })
 ```
 
+{% aside %}
+{% component-inline-prop propPath=["icon"] %}📢{% /component-inline-prop %}
```
Comment (ChrisLaneAU) [inline docs/src/content/pages/fields/slug.mdoc:29]: This has new syntax now, right? Might need to merge `main` and update this?

**Review by ChrisLaneAU — COMMENTED** [review]
Just some small suggestions but looking great 🙌  I find these demo/examples super useful when reading docs 😃

File: docs/src/components/fields/integer.tsx (hunk 4 above)
Comment (jtbrolin) [reply docs/src/components/fields/integer.tsx:18]: Haha wow 🤯

File: docs/src/components/fields/multiselect.tsx (hunk 5 above)
Comment (jtbrolin) [reply docs/src/components/fields/multiselect.tsx:20]: Ahh yeah this is cool, have applied to the multiselect and the select, was there any other place you thought the typing could be improved @ChrisLaneAU? 🙏🏻

Comment (ChrisLaneAU) [reply docs/src/components/fields/multiselect.tsx:20]: That looks good to me :)

File: docs/src/content/pages/fields/slug.mdoc (hunk 9 above)
Comment (jtbrolin) [reply docs/src/content/pages/fields/slug.mdoc]: Ahh thanks for that catch!!

File: docs/src/content/pages/fields/array.mdoc (hunk 6 above)
Comment (jtbrolin) [reply docs/src/content/pages/fields/array.mdoc]: Nah it was just written badly, I've now changed to `etc` at the end. But unsure if this section makes sense so feel free to suggest improvements if you can think of any 🙌🏻

**Review by BarnabyBishop — APPROVED** [review]
Looking good, one small comment on "Resume" for the File field but it isn't a biggy.

File: docs/src/components/fields/file.tsx [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    data: Uint8Array;
+    extension: string;
+    filename: string;
+  } | null>(null);
+
+  return (
+    <FieldDemoFrame>
+      <FileFieldInput
+        label="Resume"
+        description="The resume for this user"
```
[the remark this answers is not in the document]
Comment (jtbrolin) [reply docs/src/components/fields/file.tsx:18]: Ahh yes, I can see how this is confusing 😅 Maybe I could just use `CV` as the term, might be clearer? @BarnabyBishop

[the remark this answers is not in the document]
Comment (BarnabyBishop) [reply docs/src/components/fields/file.tsx:18]: I think Resume is more common in the US than a CV so maybe keep it as is?

[the remark this answers is not in the document]
Comment (jtbrolin) [reply docs/src/components/fields/file.tsx]: Updated the `description` now @BarnabyBishop let me know if that works 🙏🏻 :)

[the remark this answers is not in the document]
Comment (BarnabyBishop) [reply docs/src/components/fields/file.tsx]: _Very_ clear now. Nice!

**Pull request merged**